python/lsst/daf/butler/core/registry.py

```diff
@@ -182,8 +202,39 @@ def fromConfig(registryConfig, schemaConfig=None, dimensionConfig=None, create=F
             else:
                 raise ValueError("Incompatible Registry configuration: {}".format(registryConfig))
 
-        cls = doImport(registryConfig["cls"])
-        return cls(registryConfig, schemaConfig, dimensionConfig, create=create, butlerRoot=butlerRoot)
+        cls = Registry.getRegistryClass(registryConfig)
+
+        return cls(registryConfig, schemaConfig, dimensionConfig, create=create,
+                   butlerRoot=butlerRoot)
+
+    @classmethod
+    def getRegistryClass(cls, registryConfig):
```

Comment:
Reading the code here and seeing that it doesn't really need `cls`, and thinking of my comment on `getDialect` I am tending towards this also being an instance method of `RegistryConfig`.

Reply:
Moved to `RegistryConfig` as an instance method.